src/metabase/query_processor/pivot.clj

```diff
@@ -461,6 +463,196 @@
              pivot-options     (or
                                 (not-empty (select-keys query [:pivot-rows :pivot-cols]))
                                 (pivot-options query (get-in query [:info :visualization-settings])))
+             query             (assoc-in query [:middleware :pivot-options] pivot-options)
              all-queries       (generate-queries query pivot-options)
              column-mapping-fn (make-column-mapping-fn query)]
          (process-multiple-queries all-queries rff column-mapping-fn))))))
+
+;; post process utils for pivot exports
```

Comment:
I would prefer putting the export stuff in a separate namespace since the existing ~500 lines is a good size already and it's a nice logical separation of concerns between the code that runs multiple queries and the post-processing code to transform the results. 

Then we can also easily despair that there isn't a test namespace for the post-processing transformation functions 😉

Reply:
Moved to a new namespace and started adding some basic tests. Again, a follow up PR will improve coverage a bit further.